src/nbl/ext/ImGui/ImGui.cpp

```diff
@@ -775,40 +1007,39 @@ namespace nbl::ext::imgui
+			std::make_pair(bool(creationParams.transfer), "Invalid `creationParams.transfer` is nullptr!"),
+			std::make_pair(bool(creationParams.renderpass), "Invalid `creationParams.renderpass` is nullptr!"),
+			(creationParams.assetManager && creationParams.utilities && creationParams.transfer && creationParams.renderpass) ? std::make_pair(bool(creationParams.utilities->getLogicalDevice()->getPhysicalDevice()->getQueueFamilyProperties()[creationParams.transfer->getFamilyIndex()].queueFlags.hasFlags(IQueue::FAMILY_FLAGS::TRANSFER_BIT)), "Invalid `creationParams.transfer` is not capable of transfer operations!") : std::make_pair(false, "Pass valid required UI::S_CREATION_PARAMETERS!"),
+			std::make_pair(bool(creationParams.resources.texturesInfo.setIx <= 3u), "Invalid `creationParams.resources.textures` is outside { 0u, 1u, 2u, 3u } set!"),
+			std::make_pair(bool(creationParams.resources.samplersInfo.setIx <= 3u), "Invalid `creationParams.resources.samplers` is outside { 0u, 1u, 2u, 3u } set!"),
+			std::make_pair(bool(creationParams.resources.texturesInfo.bindingIx != creationParams.resources.samplersInfo.bindingIx), "Invalid `creationParams.resources.textures.bindingIx` is equal to `creationParams.resources.samplers.bindingIx`!"),
+			std::make_pair(bool(validateResourcesInfo()), "Invalid `creationParams.resources` content!")
 		});
 
 		for (const auto& [ok, error] : validation)
 			if (!ok)
 			{
-				m_creationParams.utilities->getLogger()->log(error, system::ILogger::ELL_ERROR);
+				creationParams.utilities->getLogger()->log(error, ILogger::ELL_ERROR);
 				assert(false);
 			}
 
-		smart_refctd_ptr<nbl::video::IGPUCommandBuffer> transistentCMD;
+		smart_refctd_ptr<IGPUCommandBuffer> transistentCMD;
 		{
 			using pool_flags_t = IGPUCommandPool::CREATE_FLAGS;
 
-			smart_refctd_ptr<nbl::video::IGPUCommandPool> pool = m_creationParams.utilities->getLogicalDevice()->createCommandPool(m_creationParams.transfer->getFamilyIndex(), pool_flags_t::RESET_COMMAND_BUFFER_BIT|pool_flags_t::TRANSIENT_BIT);
+			smart_refctd_ptr<IGPUCommandPool> pool = creationParams.utilities->getLogicalDevice()->createCommandPool(creationParams.transfer->getFamilyIndex(), pool_flags_t::RESET_COMMAND_BUFFER_BIT|pool_flags_t::TRANSIENT_BIT);
 			if (!pool)
 			{
-				m_creationParams.utilities->getLogger()->log("Could not create command pool!", system::ILogger::ELL_ERROR);
+				creationParams.utilities->getLogger()->log("Could not create command pool!", ILogger::ELL_ERROR);
 				assert(false);
 			}
 			
 			if (!pool->createCommandBuffers(IGPUCommandPool::BUFFER_LEVEL::PRIMARY, 1u, &transistentCMD))
 			{
-				m_creationParams.utilities->getLogger()->log("Could not create transistent command buffer!", system::ILogger::ELL_ERROR);
+				creationParams.utilities->getLogger()->log("Could not create transistent command buffer!", ILogger::ELL_ERROR);
 				assert(false);
 			}
 		}
```

Comment:
Think about this https://github.com/Devsh-Graphics-Programming/Nabla/pull/707#discussion_r1792329694

P.S. spelling: tranSIENT not transistent